src/ast/passes/semantic_analyser.cpp

```diff
@@ -1853,7 +1857,72 @@ If you're seeing errors, try clamping the string sizes. For example:
     }
     call.return_type = CreateUInt64();
   } else {
-    call.addError() << "Unknown function: '" << call.func << "'";
+    // Check here if this corresponds to an external functions. We convert the
+    // external type metadata into the internal `SizedType` representation and
+    // check that they are exactly equal.
+    bool found = false;
+    for (const btf::Function &func :
+         type_metadata_.global | btf::filter<btf::Function>()) {
+      if (func.name() != call.func ||
+          (func.linkage() != btf::Function::Linkage::Global &&
+           func.linkage() != btf::Function::Linkage::Extern)) {
+        break;
+      }
+      found = true;
+      auto proto = func.type();
+      if (!proto) {
+        call.addError() << "Unable to find function proto: "
+                        << proto.takeError();
+        continue;
+      }
+      auto return_type = proto->return_type();
+      if (!return_type) {
+        call.addError() << "Unable to read return type: "
+                        << return_type.takeError();
+        continue;
+      }
+      auto argument_types = proto->argument_types();
+      if (!argument_types) {
+        call.addError() << "Unable to read argument types: "
+                        << argument_types.takeError();
+        continue;
+      }
+      if (argument_types->size() != call.vargs.size()) {
+        call.addError() << "Expected " << argument_types->size()
+                        << " arguments, got only " << call.vargs.size();
+        continue;
+      }
+      auto compat_return_type = getCompatType(*return_type);
+      if (!compat_return_type) {
+        call.addError() << "Unable to convert return type: "
+                        << compat_return_type.takeError();
+        continue;
+      }
+      std::vector<SizedType> args;
+      for (size_t i = 0; i < argument_types->size(); i++) {
+        const auto &[name, type] = argument_types->at(i);
```

Comment:
```suggestion
        const auto &[_, type] = argument_types->at(i);
```

Reply:
I don't know how I feel about this. The analyzer will still complain about it, even if the variable is named `_`. It wants you to do `const auto &type = argument_types->at(i).second;` but I think this is ugly and less clear. I like the unpacking syntax since you mentally reason that it's a pair, and they are `[name, type]`. WDYT?

Reply:
For me it's just about reading convention, if the variable is named `_` then I know it's going to be used. But NBD either way.

Reply:
I'm with you, if the analyzer didn't complain anyways I would use `_`. I'll probably just use `name` in the updating error message anyways.